### DISCUSSION TITLE
Suggestion : Spreadsheet: show all aliases (and/or) formulas in situ

### DISCUSSION THREAD

**Discussion by hooperbloob** [FreeCAD/FreeCAD#15024] [Discussions that will turn in to issues but need refining]
While the user can hover their mouse cursor over selected cells to have any aliases pop up, there doesn't seem to be a way to view all of them at once.  Ditto for any formulas.

I'd like to see visibility 'toggles' that renders any aliases or formulas for a cell within the cell itself. Any existing values would also be shown:

```
[ 123.5 ]          [x] values   [ ] aliases  [ ] formulas
[ length ]         [ ] values   [x] aliases  [ ] formulas
[ =B3 * 2.54 ]     [ ] values   [ ] aliases  [x] formulas

[ 123.5 : length : =B3 * 2.54 ]    [x] values   [x] aliases  [x] formulas
```

Column widths could expand as needed to show non-value text and revert back to their original widths afterwards.
Non-value text could be colour-coded to tell them apart.

**Comment by krushia** [comment]
Alternate suggestion: add a `%F: formula` to this setting:
![Screenshot_20240629_131932](https://github.com/FreeCAD/FreeCAD/assets/2147138/62f8735f-eab8-45fb-be64-a0bd5fd2397b)

To get your checkboxes, you could have toolbar buttons that change the setting with a macro. Auto column width would be a separate feature.

**Comment by luzpaz** [comment]
Feel free to open a FR for this.
